### PR TITLE
Add Icons to Mention chips and update chip labels

### PR DESCRIPTION
**Pull request by thenamankumar**
This PR:

1. Adds icons to all the mention chips enabled by default namely Files, Symbols, Remote Repositories & Web URL. 
2. Renames Sourcegraph Repositories to Remote Repositories.
3. Updates the label for Symbol mention chips to only show symbol name and not the whole file path as well. 
4. Updates the level for File mention chips to only show file base path.

This PR doesn't update icons for all the other experimental openctx providers yet which are not enabled by default. For all the other providers there is a default DocumentIcon set. 

CleanShot 2024-05-30 at 23 50 18@2x

Closes CODY-1946

## Test plan

Manually tested

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -25,14 +25,14 @@
...
-    "Supercompletions"
-  ],
+  "cSpell.words": ["Supercompletion", "Supercompletions"],
   "[typescript]": {
     "editor.defaultFormatter": "biomejs.biome"
   },
   "[typescriptreact]": {
     "editor.defaultFormatter": "biomejs.biome"
   },
+  "[css]": {
```
Comment (sqs) [inline .vscode/settings.json:35]: intentional diff?

File: vscode/webviews/promptEditor/nodes/ContextItemMentionNode.module.css [hunk 2]
```diff
@@ -3,6 +3,19 @@
...
+    margin-right: 3px;
+
+
+    .icon {
+        display: inline-block;
+        xvertical-align: -1px;
+        xmargin: 0 2px;
+        width: 13px;
+        height: 13px;
+        xuser-select2: none;
```
Comment (sqs) [inline vscode/webviews/promptEditor/nodes/ContextItemMentionNode.module.css:15]: remove the lines with x prefixes

File: vscode/webviews/promptEditor/nodes/ContextItemMentionNode.ts [hunk 3]
```diff
@@ -189,12 +204,13 @@ export function contextItemMentionNodeDisplayText(contextItem: SerializedContext
...
-                return `@${contextItem.title}`
+                return contextItem.title
             }
-            return `@${decodeURIComponent(displayPath(URI.parse(contextItem.uri)))}${rangeText}`
+
+            return `${decodeURIComponent(displayPathBasename(URI.parse(contextItem.uri)))}${rangeText}`
 
         case 'symbol':
-            return `@${displayPath(URI.parse(contextItem.uri))}${rangeText}#${contextItem.symbolName}`
+            return `${contextItem.symbolName}`
```
Comment (sqs) [inline vscode/webviews/promptEditor/nodes/ContextItemMentionNode.ts:213]: remove backticks and string interpolation, just use contextItem.symbolName

**Review by sqs — APPROVED** [review]
and confirming using the arrow keys with the cursor in the input box is natural?

File: .vscode/settings.json (hunk 1 above)
Comment (thenamankumar) [reply .vscode/settings.json:35]: Yes, no default linter configured for css.

File: vscode/src/context/openctx/remoteRepositorySearch.ts [hunk 4]
```diff
@@ -24,6 +24,7 @@ const RemoteRepositorySearch: Provider & {
             return repositories.map(repo => ({
                 uri: repo.url,
                 title: repo.name,
+                description: ' ',
```
[the remark this answers is not in the document]
Comment (thenamankumar) [reply vscode/src/context/openctx/remoteRepositorySearch.ts:27]: By default we show <title> <uri>, but the repo.url is almost same as the repo name so `description: ' '` is just a hack to not show the URI.

[the remark this answers is not in the document]
Comment (eseliger) [reply vscode/src/context/openctx/remoteRepositorySearch.ts:27]: can you leave a comment on that?

Comment (eseliger) [inline vscode/src/context/openctx/remoteRepositorySearch.ts:27]: can we still get a comment here on why we put the `' '` here?

**Review by eseliger — APPROVED** [review]

File: vscode/src/context/openctx/remoteRepositorySearch.ts (hunk 4 above)
Comment (thenamankumar) [reply vscode/src/context/openctx/remoteRepositorySearch.ts:30]: done

**Comment by sqs** [comment]
Found a blocking bug that makes it impossible to add a space and start typing after the @-mention:

https://github.com/sourcegraph/cody/assets/1976/1d0a5b27-319b-443a-97a9-5ae576826046

**Pull request merged**